ResearchKit/Common/ORKDiscreteGraphView.m

```diff
@@ -0,0 +1,1023 @@
+            UILabel *axisTitleLabel = [[UILabel alloc] initWithFrame:CGRectMake(0, labelYPosition, CGRectGetWidth(self.yAxisView.frame) - kAxisMarkingRulerLength, labelHeight)];
+            
+            if (yValue != 0) {
+                axisTitleLabel.text = [NSString stringWithFormat:@"%0.0f", yValue];
+            }
+            axisTitleLabel.backgroundColor = [UIColor clearColor];
+            axisTitleLabel.textColor = self.axisTitleColor;
+            axisTitleLabel.textAlignment = NSTextAlignmentRight;
+            axisTitleLabel.font = self.isLandscapeMode ? [UIFont fontWithName:self.axisTitleFont.familyName size:16.0f] : self.axisTitleFont;
+            axisTitleLabel.minimumScaleFactor = 0.8;
+            [self.yAxisView addSubview:axisTitleLabel];
+        }
+    }
+}
+
+- (void)drawhorizontalReferenceLines {
+    [self.referenceLines removeAllObjects];
+    
+    UIBezierPath *referenceLinePath = [UIBezierPath bezierPath];
+    [referenceLinePath moveToPoint:CGPointMake(kORKGraphLeftPadding, kORKGraphTopPadding + CGRectGetHeight(self.plotsView.frame)/2)];
+    [referenceLinePath addLineToPoint:CGPointMake(CGRectGetWidth(self.frame), kORKGraphTopPadding + CGRectGetHeight(self.plotsView.frame)/2)];
+    
+    CAShapeLayer *referenceLineLayer = [CAShapeLayer layer];
+    referenceLineLayer.strokeColor = self.referenceLineColor.CGColor;
+    referenceLineLayer.path = referenceLinePath.CGPath;
+    referenceLineLayer.lineDashPattern = self.isLandscapeMode ? @[@12, @7] : @[@6, @4];
+    [self.plotsView.layer addSublayer:referenceLineLayer];
+    
+    [self.referenceLines addObject:referenceLineLayer];
+}
+
+- (void)drawVerticalReferenceLines {
+    for (int i=1; i<self.numberOfXAxisTitles; i++) {
+        
+        CGFloat positionOnXAxis = ((CGRectGetWidth(self.plotsView.frame) / (self.numberOfXAxisTitles - 1)) * i);
+        
+        UIBezierPath *referenceLinePath = [UIBezierPath bezierPath];
+        [referenceLinePath moveToPoint:CGPointMake(positionOnXAxis, 0)];
+        [referenceLinePath addLineToPoint:CGPointMake(positionOnXAxis, CGRectGetHeight(self.plotsView.frame))];
+        
+        CAShapeLayer *referenceLineLayer = [CAShapeLayer layer];
+        referenceLineLayer.strokeColor = self.referenceLineColor.CGColor;
+        referenceLineLayer.path = referenceLinePath.CGPath;
+        referenceLineLayer.lineDashPattern = self.isLandscapeMode ? @[@12, @7] : @[@6, @4];
+        [self.plotsView.layer addSublayer:referenceLineLayer];
+        
+        [self.referenceLines addObject:referenceLineLayer];
+    }
+}
+
+- (void)drawGraphForPlotIndex:(NSInteger)plotIndex {
+    [self prepareDataForPlotIndex:plotIndex];
+    
+    if ([self numberOfValidValues] > 0 && self.shouldConnectRanges) {
+        [self drawLinesForPlotIndex:plotIndex];
+    }
+    [self drawPointCirclesForPlotIndex:plotIndex];
+}
+
+- (void)drawPointCirclesForPlotIndex:(NSInteger)plotIndex {
+    CGFloat pointSize = self.isLandscapeMode ? 10.0f : 8.0f;
```

Comment:
This method is different between *line* and *discrete* but there's some code reuse opportunities as well.